### ISSUE TITLE
Recipe_History table

### ISSUE THREAD

**Issue by jeffhennen** [ZachSellman/hennen_recipes#17]
This is a lower priority where it would make a new entry into this table when a recipe is revised. This will try and tell us what recipe was revised and then for what reason.

Recipe_History
- ID
- Recipe_ID - Foreign Key of Recipe table
- New Recipe (Parent Recipe) -> Foreign Key of new recipe
- Reason(s) for revision
- Revision Date.

This will also include the use of the recipe table for the use of unlisted so we can see all or most of the revisions of a recipe.

**Comment by ZachSellman** [comment]
TL;DR this can be tracked already using the notes table.

I think this is problematic from a standpoint that every change to a recipe will create a new version. This can really quickly get out of hand if there are spelling/grammar mistakes, etc. It will be difficult retrieve and parse all past "versions" for relevant versions.

Another way to think of this is that any MAJOR changes to a recipe will fundamentally mean it is no longer the same recipe; if you change pork sausage to venison and pork sausage it's no longer the same recipe. It's either a new recipe alongside pork sausage, OR if pork sausage by itself was underwhelming, pork sausage could be renamed/edited and a note could be added as to why. Someone curious about previous iterations of the recipe could check the notes and see what changes were made when and why.

**Comment by jeffhennen** [comment]
The way I see it, the recipe when it is revised it will place a record into the recipe history table that will reference the recipe and then the follow up revised recipe and then include that note saying what the reason for the recipe change.  Also with the new recipe created it would require placing the old version as unlisted so it is not openly seen in the general recipes. Only able to be seen if you look at the primary new recipe. 

A big reason for this would be to have a good detailed record of what you tried and then also to be able to use this old version of a full recipe to then be a starting point for a new version if you did not like the newest version. This way you don't have to manually do the adjustments to get back to the old version. 

Also I don't think a new recipe should be created every single time you edit it. I am going to have some kind of selection that you can say this is a new version by means of a check box before and around the save button within the edits. This way you don't create a ton of new records for every single edit. Then I also think that would we end up having a max revisions and then it would drop records once you get maybe above 10 and then delete the oldest.

For the reason for the change could be done using a note but the notes initially are for recipe specific information which would be for the form of the recipes. For instance things to keep an eye out for when mixing, do not mix too much because it will cause cheesecake to expand too much and crack and dry out. Or just general information talking about the recipe and different things you can do with it. Just having the note I don't think in itself would be sufficient itself to be able to properly do version controls regarding recipes.

**Comment by ZachSellman** [comment]
I see what you're saying. It definitely adds some complexity regarding management of the recipes. Especially if you want to be able to set an old version to the "current" version. We'll definitely need some sort of limit just to keep things from spiraling around too much when an old recipe is brought back being the "live" version.
